src/main/scala/li/cil/oc/integration/forestry/UpgradeBeekeeper.scala

```diff
@@ -112,86 +94,34 @@ class UpgradeBeekeeper(val host: EnvironmentHost with internal.Agent) extends pr
-          slot = findEmptySlot(inv)
-
-        val u = inv.getStackInSlot(slot)
-        if (u == null)
-          inv.setInventorySlotContents(slot, upgrade)
-        else {
-          u.stackSize += upgrade.stackSize
-          inv.setInventorySlotContents(slot, u)
-        }
-        inventory.setInventorySlotContents(host.selectedSlot, null)
-        result(true)
-
-      case _ => result(false, "Not facing an industrial apiary")
-    }
+    val pos = position.offset(facing)
+    var amount = 999
+    if (args.count() > 1)
+      amount = args.checkInteger(1)
+    result(UpgradeBeekeeperUtil.addIndustrialUpgrade(pos, host.mainInventory(), host.selectedSlot(), amount))
   }
   @Callback(doc = """function(side:number, slot: number):table -- Get industrial upgrade in the given slot of the industrial apiary at the given side.""")
   def getIndustrialUpgrade(context: Context, args: Arguments): Array[AnyRef] = {
-    if (!Loader.isModLoaded("gendustry")) return result(Unit, "Gendustry not loaded!")
     val facing = checkSideForAction(args, 0)
-    position.world.get.getTileEntity(position.offset(facing)) match {
-      case ia: IIndustrialApiary =>
-        val inv = ia.asInstanceOf[ISidedInventory]
-        val slot = args.checkInteger(1) + 1
-        if (slot < 2 || slot > 5)
-          return result(Unit, "Wrong slot index (should be 1-4)")
-
-        result(inv.getStackInSlot(slot))
-
-      case _ => result(Unit, "Not facing an industrial apiary")
-    }
+    val pos = position.offset(facing)
+    val slot = args.checkInteger(1)
+    if (slot < 1 || slot > 4)
+      return result(Unit, "Wrong slot index (should be 1-4)")
+    result(UpgradeBeekeeperUtil.getIndustrialUpgrade(pos, slot))
   }
-  @Callback(doc = """function(side:number, slot: number):boolean -- Remove industrial upgrade from the given slot of the industrial apiary at the given side.""")
+  @Callback(doc = """function(side:number, slot: number[, amount: number]):boolean -- Remove industrial upgrade from the given slot of the industrial apiary at the given side.""")
   def removeIndustrialUpgrade(context: Context, args: Arguments): Array[AnyRef] = {
-    if (!Loader.isModLoaded("gendustry")) return result(false, "Gendustry not loaded!")
     val facing = checkSideForAction(args, 0)
-    position.world.get.getTileEntity(position.offset(facing)) match {
-      case ia: IIndustrialApiary =>
-        val inv = ia.asInstanceOf[ISidedInventory]
-        val slot = args.checkInteger(1) + 1
-        if (slot < 2 || slot > 5)
-          return result(false, "Wrong slot index (should be 1-4)")
-
-        val u = inv.getStackInSlot(slot)
-        val res = InventoryUtils.insertIntoInventory(u, host.mainInventory)
-        inv.setInventorySlotContents(slot, if (u.stackSize > 0) u else null)
-        result(res)
-
-      case _ => result(false, "Not facing an industrial apiary")
-    }
+    val pos = position.offset(facing)
+    val slot = args.checkInteger(1)
+    var amount = 999
```

Comment:
Please, don't use magic numbers. Create constant for storing value 64 (default maximum stack size) and use it.